### ISSUE TITLE
[manuf] consolidate DICE certs to a single flash info page

### ISSUE THREAD

**Issue by timothytrippel** [lowRISC/opentitan#22808]
Currently DICE certs are spread across 3 flash info pages. However, since the CDI certs are both modified at the ROM_EXT stage, we can consolidate them into a single page to save 2 pages worth of space.

**Comment by tommychiu-github** [comment]
Based on current flash layout, 
pg16: UdsCert
pg17: CDI_0
pg19: CDI_1

How about putting the entire DICE chain on page 19? So that we can have a whole range from page 5~18 reserving for SW?

**Comment by timothytrippel** [comment]
Sounds reasonable to me.

**Comment by tommychiu-github** [comment]
Revisit this ticket, the saving space maybe not as big as the original expectation.
Only CDI_* are varied once we update ROM_EXT and the later FW, so it's not necessary to reserve a page for them.
However the UdsCert[chain] are expected to be static values across the whole chip lifecycle, and will be access everytime when RKP requests a CSR, I think we may need one page to store the content.

Proposal:
pg18: UdsCert[chain]
pg19: entire DICE (where the UdsCert is parsed form pg18)

WDYT?

**Comment by timothytrippel** [comment]
Sorry I am not sure I understand your proposal. There are three DICE certs (UDS, CDI_0, and CDI_1). Together, they make up the "DICE chain". They are currently all placed on separate flash info pages. They are small enough, that they could all be placed on a single flash info page. How/when/if they are updated/accessed is somewhat irrelevant no? Since by the time owner firmware stage executes, this page becomes read-only. Only personalization FW and ROM_EXT ever has write access to this page (to handle initial provisioning  / fw updates respectively).

So I am confused when you say your proposal is to put "UDS cert on page 18" and the "entire DICE on page 19". Are you proposing putting the UDS cert on page 18 and the two CDI certs (0 and 1) on page 19?

**Comment by tommychiu-github** [comment]
The main goal is to have a redundent UdsCertChain (nothing to do with the CDI_*) to be placed in a separate flash page from the entire DICE chain.
For DICE, it requires all them to be composed together to form a valid chain, which is fine to be in page 19.
However for RKP, it requires not only a valide DICE chain but also an UdsCertChain which is rooted to the chip vendor's proprietary ROOT. (https://cs.android.com/android/platform/superproject/main/+/main:hardware/interfaces/security/rkp/aidl/android/hardware/security/keymint/generateCertificateRequestV2.cddl;l=64)

So to my understanding, the layout may turn to
page 18: [FabRootCert || IntermediateCert || UdsCert ],
page 19: entire DICE chain (including the same UdsCert as pg18)

**Comment by timothytrippel** [comment]
Ah thanks for clarifying @tommychiu-github .

The current reference provisioning firmware in the repo is designed to serve as a baseline for SKU-owners (i.e. various customers) to further customize to suit their needs. For example, the reference personalization firmware in the repo currently provisions three DICE certs and a TPM cert. However, as you point out, the RKP use case requires some additional certs. 

I just filed #23426, as the plan is to refactor the reference personalization firmware to be more amenable to various customizations like the one you describe to be easily added for various SKU owners. Once that lands it should be easy to add a customization that adds the cert chain your SKU may require.

As for this issue, I will close it when I consolidate the three DICE certs (UDS, CDI_0, CDI_1) into a single page, page 19. Feel free to file another issue to track the RKP customization for your SKU, if that is something you would like to contribute upstream.

**Comment by tommychiu-github** [comment]
Thank you @timothytrippel for the detailed description. I think there might be some things that I can contribute to the upstream.
